ksp_plugin_adapter/localization_extensions.cs

```diff
@@ -139,35 +143,38 @@ internal static class L10N {
   }
 
   public static string CacheFormat(string name, params string[] args) {
+    return lru_cache_.Get(name, args, () => Format(name, args));
+  }
+
+  private static string Format(string name, params string[] args) {
     // Nested <<>> rules don’t work prior to Lingoona 1.7.0 (2021-07-07).
     // We need them for some of the more advanced grammatical constructs.
     // We manually replace <<X:n>> placeholders and leave the result to
     // Lingoona. This allows for some substitution inside <<>> rules, and is
     // enough for our purposes.  <<X:n>> placeholders are substituted
     // untransformed, so the behaviour is consistent with what Lingoona 1.7.0 or
     // later would do.
-    return lru_cache_.Get(
-        name, args,
-        () => {
-          string template = Localizer.GetStringByTag(name);
-          // KSP substitutes arguments that are themselves localization keys,
-          // e.g., the names of the default vessels.
-          var resolved_args = new List<string>();
-          foreach (var arg in args) {
-            // KSP treats nulls as empty strings here (but not in the overload
-            // that takes objects).
-            resolved_args.Add(
-                arg == null ? "" : (Localizer.Tags.GetValueOrNull(arg) ?? arg));
-          }
-          for (int n = 1; n <= resolved_args.Count; ++n) {
-            template.Replace($"<<X:{n}>>", resolved_args[n - 1]);
-          }
-          // These escapes are handled by KSP *after* formatting, contrary to
-          // the \u ones and the ｢｣ for {}, which are handled when the Localizer
-          // loads the tags.
-          return Lingoona.Grammar.useGrammar(template, resolved_args)
-              .Replace(@"\n", "\n").Replace(@"\""", "\"").Replace(@"\t", "\t");
-        });
+    string template = Localizer.GetStringByTag(name);
```

Comment:
Move this variable and the comment block closer to the point of use, near lien 167.